litert/vendors/qualcomm/qnn_backend_test/builder_test/topk_test.cc

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Qualcomm Innovation Center, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <vector>
+
+#include "QnnTypes.h"  // from @qairt
+#include "litert/vendors/qualcomm/core/builders/topk_op_builder.h"
+#include "litert/vendors/qualcomm/qnn_backend_test/test_utils.h"
+
+using testing::ElementsAre;
+using testing::FloatNear;
+using testing::Pointwise;
+namespace litert::qnn {
+namespace {
+// TODO (chunhsue-qti): The following param tests will be exactly the same when
+// running on arm64 device since qnn manager will use online SoC info
+// automatically. Maybe we can find out a way to skip them.
+INSTANTIATE_TEST_SUITE_P(
+    , QnnModelTest,
+    ::testing::Values(std::make_tuple(kTestingDefaultQnnOptions, "SM8650"),
+                      std::make_tuple(kTestingDefaultQnnOptions, "SM8750"),
+                      std::make_tuple(kTestingDefaultQnnOptions, "SM8850")),
+    QnnTestPrinter);
```

Comment:
Let's implement a unified helper function in `test_utils.h` to streamline our testing process.
```c++
inline auto GetQnnModelParams() {
    return ::testing::Combine(
        ::testing::Values(kTestingDefaultQnnOptions),
        ::testing::Values("SM8650", "SM8750", "SM8850")
    );
}
```
and then we can use the following in each test.
```c++
INSTANTIATE_TEST_SUITE_P(, QnnModelTest, GetQnnDeviceParams(), QnnTestPrinter);
```